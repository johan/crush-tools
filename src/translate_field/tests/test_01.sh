test_number=01
description="translate blank values (middle field)"

infile="$test_dir/test.in"
expected="$test_dir/test_$test_number.expected"

subtest=1
outfile="$test_dir/test_$test_number.$subtest.actual"
$bin -f 2 -m "=BLANK" "$test_dir/test.in" > "$outfile"
if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number $subtest "$description (index)" FAIL
else
  test_status $test_number $subtest "$description (index)" PASS
  rm "$outfile"
fi

subtest=2
outfile="$test_dir/test_$test_number.$subtest.actual"
$bin -F Field-1 -m "=BLANK" "$test_dir/test.in" > "$outfile"
if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number $subtest "$description (label)" FAIL
else
  test_status $test_number $subtest "$description (label)" PASS
  rm "$outfile"
fi
