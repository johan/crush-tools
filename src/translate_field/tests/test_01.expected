Field-0,Field-1,Field-2
a,b,c
e,f,g
h,BLANK,j
code,in,c
code,in,pl
code,in,sh
